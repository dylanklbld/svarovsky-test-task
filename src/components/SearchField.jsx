import React, {useState} from 'react'

import useDebounce from '../hooks/useDebounce'
import useEffectExceptMount from '../hooks/useEffectExceptMount'

//  After there is a change in the field, wait for 200ms
const debounce = 200

export const SimpleSearchField = ({onSearchValueChanged}) => {
    const [searchValue,
        setSearchValue] = useState('')
    const debouncedSearchValue = useDebounce(searchValue, debounce)

    useEffectExceptMount(() => {
        debouncedSearchValue && onSearchValueChanged(debouncedSearchValue)
    }, [debouncedSearchValue])

    return <React.Fragment>
        <div className="test-search-field">
            <span>search icon</span>
            <input
                type="text"
                className="test-search-input-field"
                placeholder="Search for a product"
                value={searchValue}
                onChange={(e) => {
                setSearchValue(e.target.value)
            }}/>
            <div></div>
        </div>
    </React.Fragment>
}
