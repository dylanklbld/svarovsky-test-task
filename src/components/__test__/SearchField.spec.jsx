import "@testing-library/jest-dom"

import { fireEvent, render, wait } from '@testing-library/react'

import React from 'react'
import { SimpleSearchField } from '../SearchField'

it('should render', () => {
    const { container } = render(<SimpleSearchField onSearchRequestDone={() => { }} />)

    expect(container).toMatchSnapshot()
})


it('should call onSearchRequestDone', async () => {
    const onSearchRequestDone = jest.fn().mockName('onSearchRequestDone')
    const { container } = render(<SimpleSearchField onSearchValueChanged={onSearchRequestDone} />)
    await wait()
    expect(container).toMatchSnapshot()

    const inputField = container.querySelector('.test-search-input-field')
    fireEvent.change(inputField, { target: { value: 'kase' } })

    await new Promise(resolve => setTimeout(resolve, 300))

    expect(onSearchRequestDone).toHaveBeenCalledWith("kase")
})