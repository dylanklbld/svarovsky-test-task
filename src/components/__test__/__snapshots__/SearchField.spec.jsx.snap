// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call onSearchRequestDone 1`] = `
<div>
  <div
    class="test-search-field"
  >
    <span>
      search icon
    </span>
    <input
      class="test-search-input-field"
      placeholder="Search for a product"
      type="text"
      value=""
    />
    <div />
  </div>
</div>
`;

exports[`should render 1`] = `
<div>
  <div
    class="test-search-field"
  >
    <span>
      search icon
    </span>
    <input
      class="test-search-input-field"
      placeholder="Search for a product"
      type="text"
      value=""
    />
    <div />
  </div>
</div>
`;
